### PR TITLE
Fix: Update `--new-rule` process

### LINKED ISSUE #857: Update `--new-rule` process
Update `--new-rule` process to:

- [x] Add a `.npmrc` file containing  `package-lock=false` (see also: https://github.com/sonarwhal/sonarwhal/commit/8638385ab5070927bb05e0483dcd5368f0b41d1d)
- [x] Have `lint:md` script from `package.json` ignore `CHANGELOG.md` files (see also: https://github.com/sonarwhal/sonarwhal/commit/dd1b3710d73b5c0e60780bde02d8a4a4a0310d0d)
- [x] Update dependencies to latest versions

### PR DESCRIPTION
**Pull request by sarvaje**
## Pull request checklist

Make sure you:

- [x] Signed the [Contributor License Agreement](https://cla.js.foundation/sonarwhal/sonarwhal)
- [x] Followed the [commit message guidelines](https://sonarwhal.com/docs/contributor-guide/getting-started/pull-requests/#commit-messagess)

For non-trivial changes, please make sure you also:

- [ ] Added/Updated related documentation.
- [x] Added/Updated related tests.

### CODE REVIEW

File: packages/sonarwhal/src/lib/cli/wizards/shared-templates/package.hbs [hunk 1]
```diff
@@ -68,7 +72,7 @@
     "lint": "npm-run-all lint:*",{{#if official}}
     "lint:js": "eslint . --cache --ext js --ext md --ext ts --ignore-path ../../.eslintignore --report-unused-disable-directives",{{else}}
     "lint:js": "eslint . --cache --ext js --ext md --ext ts --report-unused-disable-directives",{{/if}}
-    "lint:md": "markdownlint *.md",
+    "lint:md": "markdownlint --ignore CHANGELOG.md *.md",
```
Comment (molant) [inline packages/sonarwhal/src/lib/cli/wizards/shared-templates/package.hbs]: There are a few things more that need to be changed in this file:

* Add `"private": true`
* Remove `"author": ""`
* Add `"build-release": "npm run clean && npm run build:assets && tsc --inlineSourceMap false --removeComments true",` if official
* Version should be `1.0.0`
* If official, `nyc` `extends` should be `../../.nycrc`

This applies to the parser's  `package.json` as well.

File: packages/sonarwhal/src/lib/utils/misc.ts [hunk 2]
```diff
@@ -344,6 +355,10 @@ const getSonarwhalPackage = () => {
     return require(path.join(__dirname, '../../../../package.json'));
 };
 
+const sameOrigin = (url1: string, url2: string): boolean => {
```
Comment (molant) [inline packages/sonarwhal/src/lib/utils/misc.ts]: Wasn't this done in another PR?

File: packages/sonarwhal/src/lib/utils/misc.ts [hunk 3]
```diff
@@ -186,6 +186,17 @@ const toCamelCase = (value: string) => {
     }, '');
 };
 
+/** Convert '-' delimitered string to pascal case name. */
+const toPascalCase = (value: string) => {
```
Comment (molant) [inline packages/sonarwhal/src/lib/utils/misc.ts]: You are adding this method that doesn't seem to be used anywhere else in this PR.

File: packages/sonarwhal/src/lib/utils/misc.ts (hunk 2 above)
Comment (alrra) [reply packages/sonarwhal/src/lib/utils/misc.ts]: Was changed on your request (see: https://github.com/sonarwhal/sonarwhal/pull/916#discussion_r176550487)

Comment (molant) [reply packages/sonarwhal/src/lib/utils/misc.ts]: OK, I missed the notification.
Then those changes shouldn't be there, right?

Comment (sarvaje) [reply packages/sonarwhal/src/lib/utils/misc.ts]: I think git mess up after a rebase, this code shouldn't be there.
For some reason, I have an extra commit in my PR (`Chore: Remove `same-origin` dependency`) and it shouldn't be there

File: packages/sonarwhal/src/lib/utils/misc.ts (hunk 3 above)
Comment (sarvaje) [reply packages/sonarwhal/src/lib/utils/misc.ts]: This is too weird, I should be using this method in new-rule.ts

Comment (sarvaje) [reply packages/sonarwhal/src/lib/utils/misc.ts]: I'm using it in the file `new-rule.ts` line 102

File: packages/sonarwhal/src/lib/cli/wizards/shared-templates/package.hbs (hunk 1 above)
Comment (sarvaje) [reply packages/sonarwhal/src/lib/cli/wizards/shared-templates/package.hbs]: done!

File: packages/sonarwhal/src/lib/cli/wizards/new-rule.ts [hunk 4]
```diff
@@ -0,0 +1,421 @@
...
+        await copyFiles(files, rulePackage.destination);
+        await generateRuleFiles(rulePackage.destination, rulePackage);
+
+        logger.log(`
+New ${rulePackage.isMulti ? 'package' : 'rule'} ${rulePackage.name} created in ${rulePackage.destination}
+
+--------------------------------------
+----          How to use          ----
+--------------------------------------
+1. Go to the folder rule-${rulePackage.normalizedName}
```
Comment (molant) [inline packages/sonarwhal/src/lib/cli/wizards/new-rule.ts]: Should we change this message if the rule is official? We should be using `yarn` in here
